File: 240305/2개씩 그룹짓기/group-of-pairs.py
n = int(input())
A = [int(x) for x in input().split()]
B = []

A.sort()


for i in range(n):
    B.append((A[i],A[len(A)-1-i]))


sum_list = []
for j in B:
    sum_list.append(sum(j))

print(max(sum_list))

File: 240420/2개씩 그룹짓기/group-of-pairs.js
const fs = require('fs')
const input = fs.readFileSync(0).toString().trim().split(`\n`)

const N = Number(input[0])
const numbers = input[1].split(' ').map(Number)

numbers.sort((a,b) => a-b)

let groups = []

for (let i = 0; i < N; i++) {
    groups.push([])
}


for (let i = 0; i < N; i ++) {
    groups[i].push(numbers[i]);
    groups[i].push(numbers[2*N - i -1])
}


let maxSum = 0

for (let j = 0; j < N; j++) {
    if (maxSum < groups[j][0] + groups[j][1]) {
        maxSum = groups[j][0] + groups[j][1]
    }
}

console.log(maxSum)